Kubernetes/legos/k8s_get_oomkilled_pods/k8s_get_oomkilled_pods.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2023 unSkript.com
+# All rights reserved.
+#
+import pprint
+from typing import Tuple, Optional
+from pydantic import BaseModel, Field
+from kubernetes import client
+from kubernetes.client.rest import ApiException
+
+
+class InputSchema(BaseModel):
+    namespace: Optional[str] = Field(
+        '',
+        description='Kubernetes Namespace Where the Service exists',
+        title='K8S Namespace',
+    )
+
+
+
+def k8s_get_oomkilled_pods_printer(output):
+    if output is None:
+        return
+    pprint.pprint(output)
+    
+
+def k8s_get_oomkilled_pods(handle, namespace: str = "") -> Tuple:
+    """k8s_get_oomkilled_pods This function returns the pods that have OOMKilled event in the container last states
+
+       :type handle: Object
+       :param handle: Object returned from the task.validate(...) function
+
+       :type namespace: str
+       :param namespace: String, K8S Namespace as python string
+
+       :rtype: Tuple Result in tuple format.
+    """
+    result = []
+    if handle.client_side_validation is not True:
+        raise ApiException(f"K8S Connector is invalid {handle}")
+
+    v1 = client.CoreV1Api(api_client=handle)
+    pods = v1.list_pod_for_all_namespaces().items
+    pods_to_check = pods
+    if namespace != "":
+        for p in pods:
+            if p.metadata.namespace == namespace:
+                pods_to_check = p
```

Comment:
This should have been append()

Reply:
AI suggested: 

// given a kubernetes namespace, list all pods in that namespace, if no valid namespace is given, list all pods. write this in python

```
from kubernetes import client, config

def list_oomkilled_pods_in_namespace(namespace=None):
    # Loads kubeconfig from default location (~/.kube/config)
    # or KUBECONFIG environment variable.
    config.load_kube_config()

    v1 = client.CoreV1Api()
    
    if namespace:
        try:
            print(f"Listing OOMKilled pods with namespace {namespace}:")
            pods = v1.list_namespaced_pod(namespace)
        except client.rest.ApiException as e:
            print(f"Exception when calling CoreV1Api->list_namespaced_pod: {e}\nInvalid namespace")
            return
    else:
        print("Listing all OOMKilled pods:")
        pods = v1.list_pod_for_all_namespaces()

    for pod in pods.items:
        for status in pod.status.container_statuses:
            if status.state.terminated and status.state.terminated.reason == 'OOMKilled':
                print(f"Pod: {pod.metadata.name}, Namespace: {pod.metadata.namespace}, Container: {status.name}")

# Replace 'my-namespace' with your namespace
# To list all pods, just call the function with no arguments
list_oomkilled_pods_in_namespace('my-namespace')
```

Reply:
Changed the code to get all namespaces.